### PR TITLE
Added hostname-strict-https option

### LINKED ISSUE #193: Unable to set hostname-strict-https=false
##### SUMMARY

I am try to deploy middleware_automation.keycloak.keycloak_quarkus with the following playbook:
```yaml
---
- name: Playbook for Keycloak Hosts
  hosts: all
  vars:
    keycloak_quarkus_proxy_mode: edge
    keycloak_quarkus_http_enabled: true
    keycloak_quarkus_hostname_strict: false
   keycloak_quarkus_hostname_strict_backchannel: false
    
    keycloak_quarkus_admin_pass: "remembertochangeme"
    keycloak_quarkus_configure_firewalld: true
  roles:
    - middleware_automation.keycloak.keycloak_quarkus
```

But when I try to connect using HTTP, its admin page keeps blank with a loading animation. To make it work I have to manually put `hostname-strict-https=false` and restart the keycloak service.

Therefore, I am requesting to add `hostname-strict-https` to the role variables.

##### ISSUE TYPE
 - Add role variable `hostname-strict-https = false | true [default]`

### PR DESCRIPTION
**Pull request by InfoSec812**
Resolves #193 

Added a new option `keycloak_quarkus_hostname_strict_https` with no default value.

If the option is set, the Jinja template for `keycloak.conf.j2` checks to see if the value is either `true` or `false` and applies it appropriately

### CODE REVIEW

File: roles/keycloak_quarkus/templates/keycloak.conf.j2 [hunk 1]
```diff
@@ -10,6 +10,10 @@ db-password={{ keycloak_quarkus_db_pass }}
 {% endif %}
 {% endif %}
 
+{% if keycloak_quarkus_hostname_strict_https -%}
+hostname-strict-https={{ keycloak_quarkus_hostname_strict_https }}
```
Comment (guidograzioli) [inline roles/keycloak_quarkus/templates/keycloak.conf.j2:14]: you need `| lower` or python will emit False instead of false, which quarkus doesn't like

Comment (guidograzioli) [reply roles/keycloak_quarkus/templates/keycloak.conf.j2:14]: plus, please add the role parameter to meta/argument_specs.yml, with `required: no` and telling in the description that it only applies when `keycloak_quarkus_frontend_url` is not set

**Review by guidograzioli — CHANGES_REQUESTED** [review]
just a minor change needed, thanks for the contribution

File: roles/keycloak_quarkus/templates/keycloak.conf.j2 [hunk 2]
```diff
@@ -10,6 +10,10 @@ db-password={{ keycloak_quarkus_db_pass }}
 {% endif %}
 {% endif %}
 
+{% if keycloak_quarkus_hostname_strict_https -%}
```
Comment (guidograzioli) [inline roles/keycloak_quarkus/templates/keycloak.conf.j2:13]: `if keycloak_quarkus_hostname_strict_https is defined`

**Comment by InfoSec812** [comment]
@guidograzioli I modified it to better handle ensuring that the value is a boolean and it should address your concern as well.

**Comment by guidograzioli** [comment]
Thanks for updating, but the specification for the new parameter is still needed here:
https://github.com/ansible-middleware/keycloak/blob/main/roles/keycloak_quarkus/meta/argument_specs.yml#L335

**Comment by InfoSec812** [comment]
@guidograzioli OK, I think that latest change should cover it.

Thanks!

**Review by guidograzioli — APPROVED** [review]
LGTM, thanks

**Pull request merged**